### PR TITLE
Docs/0.8.0

### PR DESCRIPTION
**Pull request by drewbanin**

### CODE REVIEW

File: docs/guide/configuring-models.md [hunk 1]
```diff
@@ -75,6 +75,10 @@ Incremental models are a powerful feature in production dbt deployments. Frequen
 
 It's highly recommended to use incremental models rather than table models in production whenever the schema allows for it. This will minimize your model build times and minimize the use of database resources.
 
+There are generally two ways to configure incremental models: Simple and Advanced. While the "simple" incremental models only require a configuration change (and not a code change), they are generally less performant than an "advanced" incremental model setup. First, understand the process of setting up a simple incremental model. Then, read on to understand how to tune your incremental models for optimal performance.
+
+Incremental models require two configuration options: `sql_where` and `unique_key`.
```
Comment (jthandy) [inline docs/guide/configuring-models.md]: unique_key is not required

File: docs/guide/configuring-models.md [hunk 2]
```diff
@@ -85,19 +89,83 @@ WHERE [source].session_end_timestamp >= (select max(session_end_timestamp) from
...
 
-#### unique_key
+### Advanced incremental model usage
 
-`unique_key` is an optional parameter that specifies uniqueness on this table. Records matching this UK that are found in the table will be deleted before new records are inserted. Functionally, this allows for modification of existing rows in an incremental table. `unique_key` can be any valid SQL expression, including a single field, or a function. A common use case is concatenating multiple fields together to create a single unique key, as such: `user_id || session_index`.
+Simple incremental models blindly apply the `sql_where` filter to the entire model SELECT query. Depending on the complexity of the SQL in the model, the database planner may be able to optimize the number of records it scans while executing your query. Generally though, the database will build the entire model, then filter the modeled dataset. This can take as long, or in some cases _longer_ than a simple `table` materialization! Advanced incremental model usage involves adding a few extra lines of code to your model to ensure that only new or changed data is processed during the dbt run.
```
Comment (jthandy) [inline docs/guide/configuring-models.md]: *query planner

File: docs/guide/configuring-models.md [hunk 3]
```diff
@@ -85,19 +89,83 @@ WHERE [source].session_end_timestamp >= (select max(session_end_timestamp) from
...
+),
+
+-- Filter out only the events that have arrived since events have last been processed
+new_events as (
+
+    select *
+    from all_events
+
+    -- This is executed just before the model is executed and returns either True or False
+    {% if already_exists(this.schema, this.table) %}
```
Comment (jthandy) [inline docs/guide/configuring-models.md]: I'm pretty sure you have to use `adapter.already_exists`, right?

File: docs/guide/configuring-models.md (hunk 1 above)
Comment (drewbanin) [inline docs/guide/configuring-models.md]: :+1:

File: docs/guide/configuring-models.md (hunk 2 above)
Comment (drewbanin) [inline docs/guide/configuring-models.md]: :)

File: docs/guide/configuring-models.md (hunk 3 above)
Comment (drewbanin) [inline docs/guide/configuring-models.md]: nope, `already_exists` (and actually all the adapter wrapped funcs) are in the global namespace!

**Pull request merged**